### PR TITLE
feat: Adds Icon documentation to doc site

### PR DESCRIPTION
**Pull request by zero0Halo**
feat: Added new IconList component(s) to the doc site.

The components in IconList can render all all icons and their names, and also the number of icons available

feat: Added copy to the icon.mdx file

fix: Assigned the keys of icons to a variable

There were two places in the IconList component where I was running Object.keys(). Making it DRY.

chore: Added a large comment block to pojo.js to explain its purpose

feat: Added more styles to the li component using Typography and styled components

chore: Modified the collapse documentation's styling to more closely resemble Icon's

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [ ] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config

---------
**Outline your feature or bug-fix below**

This adds documentation for the Icon component, as well as updates the Collapse component to more closely resemble Icon's formatting.

### CODE REVIEW

File: docs/src/components/IconList/IconList.component.tsx
Comment (bjohnsonrmn) [inline docs/src/components/IconList/IconList.component.tsx:36]: When I build the doc site locally, I'm seeing the icon displayed outside of the `div` because of the `bottom-margin` of the `h6` element. You could probably fix this either by using a different element for the title or increasing the height of the `cell` class.

File: docs/src/pages/components/icon.mdx
Comment (bjohnsonrmn) [inline docs/src/pages/components/icon.mdx:14]: This is a nice touch 👍

File: docs/src/pages/components/collapse.mdx
Comment (bjohnsonrmn) [inline docs/src/pages/components/collapse.mdx]: Small typo here: `componnent`

**Review by bjohnsonrmn — COMMENTED** [review]
@zero0Halo this LGTM. Good work on thorough documentation 💯 and the live code editor is cool. 

@brandonlilly I just wanted to  call out that I'm seeing this warning when building the doc site locally. Do you think anything needs to be done about that in this ticket?  

```
 WARNING  Compiled with 3 warnings                                                                                                                                                                                                    11:26:10

 warning  in ../src/theme/system.theme.ts

"export 'MarginProps' was not found in '@xstyled/system'

 warning  in ../src/theme/system.theme.ts

"export 'PaddingProps' was not found in '@xstyled/system'

 warning  in ../src/theme/system.theme.ts

"export 'SpaceProps' was not found in '@xstyled/system'
```

**Comment by bjohnsonrmn** [comment]
@zero0Halo your updates LGTM, pending approval of @brandonlilly that this PR didn't somehow break `styled-system`.

File: docs/package.json
Comment (sharkparty) [inline docs/package.json:11]: 👍

File: docs/src/components/IconList/pojo.ts
Comment (sharkparty) [inline docs/src/components/IconList/pojo.ts]: Does `import * as Icons` not allow you to iterate through that object to perform this same operation?

File: docs/src/components/Layout/Page/Page.component.tsx
Comment (sharkparty) [inline docs/src/components/Layout/Page/Page.component.tsx:211]: This can just be `<Typography mb={1}  />` can't it?

File: docs/src/components/IconList/pojo.ts
Comment (zero0Halo) [reply docs/src/components/IconList/pojo.ts]: I'll have to ask you IRL for clarification on this because that's what this file is for, pulling in all the icons and doing an export so it can be pulled in as `import * as Icons`.

File: docs/src/components/Layout/Page/Page.component.tsx
Comment (zero0Halo) [reply docs/src/components/Layout/Page/Page.component.tsx:211]: The component that is returned needs to be wrapped in an < li > tag, so at a minimum what would have to be returned is `<li><Typography mb={1}...`. That said, because what I'm trying to pad is the container holding the content I'd prefer putting the padding on the parent rather than the child.

File: docs/src/components/IconList/pojo.ts
Comment (brandonlilly) [reply docs/src/components/IconList/pojo.ts]: You can't 
```js
import * as Icons from '@retailmenot/anchor'
```
Without grabbing every other nonicon component too.

File: docs/src/components/IconList/pojo.ts
Comment (brandonlilly) [inline docs/src/components/IconList/pojo.ts]: We should at least rename this file. `pojo` is too generic.

**Comment by brandonlilly** [comment]
> @brandonlilly I just wanted to call out that I'm seeing this warning when building the doc site locally. Do you think anything needs to be done about that in this ticket?
> 
> ```
>  WARNING  Compiled with 3 warnings                                                                                                                                                                                                    11:26:10
> 
>  warning  in ../src/theme/system.theme.ts
> 
> "export 'MarginProps' was not found in '@xstyled/system'
> 
>  warning  in ../src/theme/system.theme.ts
> 
> "export 'PaddingProps' was not found in '@xstyled/system'
> 
>  warning  in ../src/theme/system.theme.ts
> 
> "export 'SpaceProps' was not found in '@xstyled/system'
> ```

This seems to only happen with exported typescript interfaces, but tbh I don't understand the issue myself. I remember @zero0Halo changing `src/Button/index.ts` when this happened with `ButtonProps` but iirc this may have caused some other issues. 

In summary, ¯\\\_(ツ)\_/¯.

**Comment by zero0Halo** [comment]
In that case I think it was flat-out errors in the build or the usage and not just warnings. It might be something to make a ticket on but it's doubtful that's its specific to what's going on here (I think).

File: docs/src/components/Layout/Page/Page.component.tsx
Comment (sharkparty) [reply docs/src/components/Layout/Page/Page.component.tsx:211]: 👍

**Review by sharkparty — APPROVED** [review]
LGTM

**Review by brandonlilly — APPROVED** [review]

**Review by bjohnsonrmn — APPROVED** [review]
LGTM

**Pull request merged**
